lib/posthog/feature_flags.rb

```diff
@@ -118,33 +131,71 @@ def get_feature_flag(key, distinct_id, groups = {}, person_properties = {}, grou
     end
 
     def get_all_flags(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
-      # returns a string hash of all flags
+    # returns a string hash of all flags
+      response = get_all_flags_and_payloads(distinct_id, groups, person_properties, group_properties, only_evaluate_locally)
+      flags = response[:featureFlags]
+    end
 
-      # make sure they're loaded on first run
+    def get_feature_payloads(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
```

Comment:
we don't need this function

Reply:
ah, I see how it's used.

Here's a proposal for cleaner code:

Just like we did for all_flags_and_payloads, create a `get_feature_flag_and_payload` function, which has all the logic for `get_feature_flag`, and returns the flag value, payload value, and if it was evaluated locally.

Then, use this function for both `get_feature_flag`, and `get_feature_flag_payload`

The crux here is that everything comes from decide anyway, so it doesn't make sense to artificially split these out, and then have to toggle only_evaluate_locally to first see if it was locally available and then make the decide call later, when `get_feature_flag` already handles this logic.

Reply:
yep looks like I had a duplicate method named the same.. removed it 👍